### DISCUSSION TITLE
How to create session in NextJS app to bridge into a Remix app?

### DISCUSSION THREAD

**Discussion by andrewgadziksonos** [remix-run/remix#1974] [Q&A] [answered]
Hello Remix'ers,

We have a "legacy" NextJS application where we want to slowly adapt Remix over time.  Unfortunately, we need to create a session cookie in our NextJS app so that it bridges the session to a Remix app.  Has anyone tried this before?  Are we able to pull in the `create-session` code from Remix into a NextJS app?

Thank you in advance!

**Comment by sergiodxa** [comment, marked as answer]
Hey! I’m doing this right now at daffy.org

What I did was to move the auth to Remix and use the session storage and cookies from Remix package.

Because getSession expects the Cookie header in Remix I could read from `request.headers.get("cookie")` and in Next I use `request.headers.cookie` (in Next the headers is a plain object).

Then I do the login flow on Remix (I used Remix Auth) and both can read the session data using the getSession function.

That’s how I’m migrating our Next app to Remix.

**Reply by andrewgadziksonos** [reply]
Perfect, thank you @sergiodxa!
